src/components/Avatar.jsx

```diff
@@ -36,7 +50,7 @@ const Avatar = ({
     colorList = AVATAR_COLORS
 }) => {
     const userInitials = getInitials(user.name) || getInitials(user.email)
-    const avatarSize = AVATAR_SIZES.includes(size) ? size : 'l'
+    const avatarSize = size && AVATAR_SIZES.includes(size) ? size : 'l'
```

Comment:
The check is required because array of non-undefined can't include undefined:
> Type 'undefined' is not assignable to type '"xxs" | "xs" | "s" | "m" | "l" | "xl" | "xxl" | "xxxl"'.ts(2345)

Reply:
I wasn't referring to the added `size &&`. I was referring to the fact that we can remove this line altogether. What this line is doing (checking that the provided `size` prop is one of the accepted valid values) is already provided to us by the type checking mechanism.

Reply:
Ah. But this is also untrue because of https://medium.com/@martin_hotell/react-typescript-and-defaultprops-dilemma-ca7f81c661c7

Reply:
That article refers to when you have class components, but in function components there's no need to use `defaultProps`. The default values for props are better expressed in destructuring right in the arguments, as done in this component.

Reply:
Try with `strict: true` and you'll see what I mean. I don't see how this could be fixed.